etc/mpv-powershell-completion.ps1

```diff
@@ -0,0 +1,285 @@
+
+        if ($commandName -match '(--.+-file.*=)(.+)')
+        {
+            return (Get-ChildItem -file).FullName | Resolve-Path -Relative |
+                Where-Object { $_.ToLower().Contains($Matches[2].ToLower()) } |
+                ForEach-Object { if ($_.Contains(' ')) { $Matches[1] + "'$_'" } else { $Matches[1] + $_ } }
+        }
+
+        $shortCommandName = $commandName.Substring(2)
+
+        $argName = ''
+
+        if ($commandName.EndsWith('='))
+        {
+            $shortCommandName = $shortCommandName.Substring(0, $shortCommandName.Length -1)
+        }
+        elseif ($commandName.Contains('='))
+        {
+            $shortCommandName = $shortCommandName.Substring(0, $shortCommandName.IndexOf('='))
+            $argName = $commandName.Substring($commandName.IndexOf('=') + 1)
+        }
+
+        foreach ($it in $UpdateOptions)
+        {
+            if ($shortCommandName -eq $it)
+            {
+                Update-Option $it
+                break
+            }
+        }
+
+        $results = New-Object Collections.Generic.List[Object]
+
+        $exactMatches = $Options | Where-Object { $_.name -eq $shortCommandName }
+
+        foreach ($it in $exactMatches)
+        {
+            if (-not $commandName.Contains('='))
+            {
+                continue
+            }
+
+            $arguments = $null
+
+            if ($it.type -eq 'flag')
+            {
+                $arguments = 'yes', 'no'
+            }
+
+            if ($it.type -eq 'choice' -and $null -ne $it.choices)
+            {
+                $arguments = $it.choices
+            }
+
+            if ($null -ne $arguments)
+            {
+                foreach ($arg in $arguments)
+                {
+                    if ($argName -ne '')
+                    {
+                        if ($arg.StartsWith($argName))
+                        {
+                            $results.Add('--' + $it.name + '=' + $arg)
+                        }
+                    }
+                    else
+                    {
+                        $results.Add('--' + $it.name + '=' + $arg)
+                    }
+                }
+            }
+        }
+
+        if (-not $commandName.Contains('='))
+        {
+            $partlyMatches = $Options | Where-Object { $_.name.StartsWith($shortCommandName) }
+
+            foreach ($it in $partlyMatches)
+            {
+                if ($it.name -eq $shortCommandName)
+                {
+                    continue
+                }
+
+                $results.Add('--' + $it.name)
+            }
+        }
+
+        return $results
+    }
+    elseif ($commandName -eq '')
+    {
+        return (Get-ChildItem).FullName | Resolve-Path -Relative
+    }
+    else
+    {
+        return (Get-ChildItem).FullName | Resolve-Path -Relative |
+            Where-Object { $_.ToLower().Contains($commandName.ToLower()) }
+    }
+}
+
+Register-ArgumentCompleter -Native -CommandName mpv -ScriptBlock {
```

Comment:
We should allow user to unregister too.

Reply:
It's installed by dot sourcing and uninstalled by undoing the dot sourcing.

I don't know if unregister is possible or helpful, just removing the dot sourcing seem to be sufficient.